lib/hm_crypto/container.ex

```diff
@@ -137,15 +142,14 @@ defmodule HmCrypto.Container do
           secure_command,
           Crypto.private_key(),
           HmCrypto.AccessCertificate.access_certificate_binary() | HmCrypto.Crypto.public_key()
-        ) :: {:ok, command} | {:error, disclose_error}
+        ) :: {:ok, command} | {:error, disclose_error} | {:error, container_parser_error}
   def disclose(container_data, private_key, access_certificate) do
-    %{command: command, encrypted_flag: encrypted_flag?, target_serial: _, nonce: nonce} =
-      destruct_container(container_data)
-
-    if encrypted_flag? == 0x00 do
-      {:error, :unencrypted_command}
-    else
-      disclose_command(command, private_key, access_certificate, nonce)
+    with {:ok, container} <- destruct_container(container_data) do
```

Comment:
WDYT if we use [tuple-wrapping](https://medium.com/@tylerpachal/using-tuple-wrapping-to-improve-with-expressions-in-elixir-f491b06265fa) here instead of this nested conditional?

If not we could also try to use `case` instead of `with` since there's only one item being asserted and maybe even extract the `if` to a private function.

Same would apply to `disclose_error/1` on line ~174.

It might look something like the following.
```elixir
def disclose(container_data, private_key, access_certificate) do
  with {:ok, container} <- destruct_container(container_data),
       {:flag, 0x00} <- {:flag, container.encrypted_flag} do
    disclose_command(container.command, private_key, access_certificate, container.nonce)
  else
     {:flag, _} -> {:error, :unencrypted_command}
     error -> error
  end
end
```

We might also try to use some private functions since we do this `0x00` comparison in more than one place.
```elixir
defp encrypted?(0x00), do: {:ok, :encrypted}
defp encrypted?(0x00), do: {:ok, :not_encrypted}
```

Reply:
```
     error -> error
```
needs to access `container` which is the output of the first line in `with` (line 147) so can't replace `if` and move it to `with`

I can also move  `if` logic it to a function 
```elixir
def ... do
    with {:ok, container} <- destruct_container(container_data) do
      disclose_command_if_encrypted(container, private_key, access_certificate)
    end
end

def disclose_command_if_encrypted(container, private_key, access_certificate) do
    if container.encrypted_flag == 0x00 do
      {:error, :unencrypted_command}
    else
      disclose_command(container.command, private_key, access_certificate, container.nonce)
    end
end
```

It doesn't look good as it should. what do you think?